src/levanter/layers/attention.py

```diff
@@ -151,65 +152,127 @@ def dot_product_attention(
     if scaling_factor is None:
         scaling_factor = 1 / math.sqrt(query.resolve_axis(Key).size)
 
+    attention_out = None
+
     match attn_backend:
         case AttentionBackend.NVTE:
-            attention_out = _try_te_attention(
-                QPos,
-                KPos,
-                Key,
-                query,
-                key,
-                value,
-                mask,
-                bias,
-                dropout,
-                inference,
-                prng=prng,
-                attention_dtype=attention_dtype,
-                precision=precision,
-                flash_block_size=flash_block_size,
-                force_te=not was_default,
-                scaling_factor=scaling_factor,
-                logits_soft_cap=logits_soft_cap,
-            )
+            if attn_sink is None:
```

Comment:
imho slightly better to push this into _try_te_attention and let it emit a warning as applicable. In particular, the contract should be that if a backend was forced in config (e.g. the user specified NVTE), it ought to raise if it's not compatible